### PR TITLE
Optimize route sealing

### PR DESCRIPTION
**Pull request by jrudolph**
Smallish optimization for base routing infrastructure.

### CODE REVIEW

File: akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala [hunk 1]
```diff
@@ -261,6 +269,7 @@ object RejectionHandler {
       .handle { case ValidationRejection(msg, _) ⇒ complete((BadRequest, msg)) }
       .handle { case x ⇒ sys.error("Unhandled rejection: " + x) }
       .handleNotFound { complete((NotFound, "The requested resource could not be found.")) }
+      .asDefault()
```
Comment (jonas) [inline akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala]: Not sure if you consider to just change how the builder is instantiated.

```scala
   final val default =
     new Builder(isDefault = true)
     ...
```

Comment (jrudolph) [inline akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala]: Yeah, good point, that should suffice for that single use case of the flag.

Comment (jrudolph) [inline akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala]: Yes, the single line change looks much better ;) Thanks for the hint.

File: akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala [hunk 2]
```diff
@@ -37,11 +37,9 @@ object Route {
...
                          exceptionHandler: ExceptionHandler = null): Route = {
     import directives.ExecutionDirectives._
-    handleExceptions(ExceptionHandler.seal(exceptionHandler)) {
-      handleRejections(rejectionHandler.seal) {
-        route
-      }
-    }
+    // optimized as this is the root handler for all akka-http applications
+    (handleExceptions(ExceptionHandler.seal(exceptionHandler)) & handleRejections(rejectionHandler.seal))
+      .tapply(_ ⇒ route) // execute above directives eagerly, avoiding useless laziness of Directive.addByNameNullaryApply
```
Comment (ktoso) [inline akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala]: nice one, do you remember how much it showed up in profiling?

**Comment by ktoso** [comment]
LGTM

File: akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala (hunk 2 above)
Comment (jrudolph) [inline akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala]: Quite a lot in the techempower example. But the bigger share is that marshalling is still somewhat slow.

**Pull request merged**